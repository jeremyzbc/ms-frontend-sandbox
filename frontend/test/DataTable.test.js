import React from 'react';
import { mount } from 'enzyme';
import DataTable from 'frontend/components/DataTable';

const mockColumns = [
  {
    heading: 'ID',
    keyName: 'id',
    sortable: false,
    width: '25%'
  },
  {
    heading: 'Name',
    keyName: 'name',
    type: 'string',
    width: '25%'
  },
  {
    heading: 'Score',
    keyName: 'score',
    type: 'number',
    width: '25%'
  },
  {
    heading: 'Date of birth',
    keyName: 'dob',
    type: 'date',
    width: '25%'
  }
];

const mockData = [
  {
    id: 1,
    name: 'Jack',
    dob: '1990-02-02',
    score: 200
  },
  {
    id: 2,
    name: 'Jack',
    dob: '1980-12-02',
    score: 200
  },
  {
    id: 3,
    name: 'Joe',
    dob: '1996-09-22',
    score: 100
  },
  {
    id: 4,
    name: 'Simon',
    dob: '1990-02-02',
    score: 400
  },
  {
    id: 5,
    name: 'Abraham',
    dob: '1988-05-11',
    score: 400
  }
];

const createDataTable = ({
  columns = mockColumns,
  source = mockData,
  loading = false,
  error = false
} = {}) => (
  <DataTable
    columns={columns}
    source={source}
    loading={loading}
    error={error}
  />
);

describe('Test DataTable component snapshots', () => {
  it('should match snapshot of loading', () => {
    const wrapper = mount(createDataTable({ loading: true }));
    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot of error', () => {
    const wrapper = mount(createDataTable({ error: true }));
    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot of data fetched', () => {
    const wrapper = mount(createDataTable());
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Test DataTable component features', () => {
  it('should show loading text', () => {
    const wrapper = mount(createDataTable({ loading: true }));
    const text = wrapper.find('h5').text();
    expect(text).toBe('Loading...');
  });
  it('should show error text', () => {
    const wrapper = mount(createDataTable({ error: true }));
    const text = wrapper.find('h5').text();
    expect(text).toContain('something went wrong');
  });
  it('should show no records found when data length is 0', () => {
    const wrapper = mount(createDataTable({ source: [] }));
    const text = wrapper.find('h5').text();
    expect(text).toContain('No records');
  });
  it('should show table with original order', () => {
    const wrapper = mount(createDataTable());
    mockData.forEach((row, index) => {
      const id = wrapper
        .find('div.d-flex')
        .at(index + 1)
        .find('span')
        .at(0)
        .text();
      expect(id).toBe(`${row.id}`);
    });
  });
  it('should sort with asc order after first click', () => {
    const wrapper = mount(createDataTable());
    // name sorter is clicked
    const sorter = wrapper.find('.flex-column').at(0);
    sorter.simulate('click');

    let ids = [];
    for (let i = 1; i <= mockData.length; i++) {
      const id = wrapper
        .find('div.d-flex')
        .at(i)
        .find('span')
        .at(0)
        .text();
      ids.push(id);
    }
    expect(ids[0]).toBe('5'); // Abraham
    expect(ids[1]).toBe('1'); // Jack
    expect(ids[2]).toBe('2'); // Jack
    expect(ids[3]).toBe('3'); // Joe
    expect(ids[4]).toBe('4'); // Simon
  });
  it('should sort with desc order after second click', () => {
    const wrapper = mount(createDataTable());
    // name sorter is clicked
    const sorter = wrapper.find('.flex-column').at(0);
    sorter.simulate('click');
    sorter.simulate('click');

    let ids = [];
    for (let i = 1; i <= mockData.length; i++) {
      const id = wrapper
        .find('div.d-flex')
        .at(i)
        .find('span')
        .at(0)
        .text();
      ids.push(id);
    }
    expect(ids[0]).toBe('4'); // Simon
    expect(ids[1]).toBe('3'); // Joe
    expect(ids[2]).toBe('1'); // Jack
    expect(ids[3]).toBe('2'); // Jack
    expect(ids[4]).toBe('5'); // Abraham
  });
  it('should show table with original order after third click', () => {
    const wrapper = mount(createDataTable());
    // name sorter is clicked
    const sorter = wrapper.find('.flex-column').at(0);
    sorter.simulate('click');
    sorter.simulate('click');
    sorter.simulate('click');

    mockData.forEach((row, index) => {
      const id = wrapper
        .find('div.d-flex')
        .at(index + 1)
        .find('span')
        .at(0)
        .text();
      expect(id).toBe(`${row.id}`);
    });
  });
  it('should sort with multiple criterias after ctrl + clicked multiple sorter', () => {
    const wrapper = mount(createDataTable());
    // name sorter is clicked
    const nameSorter = wrapper.find('.flex-column').at(0);
    const dobSorter = wrapper.find('.flex-column').at(1);
    const scoreSorter = wrapper.find('.flex-column').at(2);
    nameSorter.simulate('click', { ctrlKey: true });
    dobSorter.simulate('click', { ctrlKey: true });
    scoreSorter.simulate('click', { ctrlKey: true });

    let ids = [];
    for (let i = 1; i <= mockData.length; i++) {
      const id = wrapper
        .find('div.d-flex')
        .at(i)
        .find('span')
        .at(0)
        .text();
      ids.push(id);
    }
    expect(ids[0]).toBe('5'); // Abraham
    expect(ids[1]).toBe('2'); // Jack
    expect(ids[2]).toBe('1'); // Jack
    expect(ids[3]).toBe('3'); // Joe
    expect(ids[4]).toBe('4'); // Simon
  });
});
