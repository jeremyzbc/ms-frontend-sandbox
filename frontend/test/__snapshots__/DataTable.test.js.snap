// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DataTable component snapshots should match snapshot of data fetched 1`] = `
<DataTable
  columns={
    Array [
      Object {
        "heading": "ID",
        "keyName": "id",
        "sortable": false,
        "width": "25%",
      },
      Object {
        "heading": "Name",
        "keyName": "name",
        "type": "string",
        "width": "25%",
      },
      Object {
        "heading": "Score",
        "keyName": "score",
        "type": "number",
        "width": "25%",
      },
      Object {
        "heading": "Date of birth",
        "keyName": "dob",
        "type": "date",
        "width": "25%",
      },
    ]
  }
  error={false}
  loading={false}
  source={
    Array [
      Object {
        "dob": "1990-02-02",
        "id": 1,
        "name": "Jack",
        "score": 200,
      },
      Object {
        "dob": "1980-12-02",
        "id": 2,
        "name": "Jack",
        "score": 200,
      },
      Object {
        "dob": "1996-09-22",
        "id": 3,
        "name": "Joe",
        "score": 100,
      },
      Object {
        "dob": "1990-02-02",
        "id": 4,
        "name": "Simon",
        "score": 400,
      },
      Object {
        "dob": "1988-05-11",
        "id": 5,
        "name": "Abraham",
        "score": 400,
      },
    ]
  }
>
  <DataTableHeader
    columns={
      Array [
        Object {
          "heading": "ID",
          "keyName": "id",
          "sortable": false,
          "width": "25%",
        },
        Object {
          "heading": "Name",
          "keyName": "name",
          "type": "string",
          "width": "25%",
        },
        Object {
          "heading": "Score",
          "keyName": "score",
          "type": "number",
          "width": "25%",
        },
        Object {
          "heading": "Date of birth",
          "keyName": "dob",
          "type": "date",
          "width": "25%",
        },
      ]
    }
    onChangeSorting={[Function]}
    sortCriterias={Array []}
  >
    <section
      style={
        Object {
          "fontWeight": 500,
          "padding": "1rem 0",
        }
      }
    >
      <div
        className="d-flex"
      >
        <span
          className="d-flex justify-content-between align-items-center"
          key="id"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          <span>
            ID
          </span>
        </span>
        <span
          className="d-flex justify-content-between align-items-center"
          key="name"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          <span>
            Name
          </span>
          <span
            className="d-inline-flex flex-column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <ArrowUpIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 1184q0 13-10 23l-50 50q-10 10-23 10t-23-10l-393-393-393 393q-10 10-23 10t-23-10l-50-50q-10-10-10-23t10-23l466-466q10-10 23-10t23 10l466 466q10 10 10 23z"
                />
              </svg>
            </ArrowUpIcon>
            <ArrowDownIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 736q0 13-10 23l-466 466q-10 10-23 10t-23-10l-466-466q-10-10-10-23t10-23l50-50q10-10 23-10t23 10l393 393 393-393q10-10 23-10t23 10l50 50q10 10 10 23z"
                />
              </svg>
            </ArrowDownIcon>
          </span>
        </span>
        <span
          className="d-flex justify-content-between align-items-center"
          key="score"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          <span>
            Score
          </span>
          <span
            className="d-inline-flex flex-column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <ArrowUpIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 1184q0 13-10 23l-50 50q-10 10-23 10t-23-10l-393-393-393 393q-10 10-23 10t-23-10l-50-50q-10-10-10-23t10-23l466-466q10-10 23-10t23 10l466 466q10 10 10 23z"
                />
              </svg>
            </ArrowUpIcon>
            <ArrowDownIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 736q0 13-10 23l-466 466q-10 10-23 10t-23-10l-466-466q-10-10-10-23t10-23l50-50q10-10 23-10t23 10l393 393 393-393q10-10 23-10t23 10l50 50q10 10 10 23z"
                />
              </svg>
            </ArrowDownIcon>
          </span>
        </span>
        <span
          className="d-flex justify-content-between align-items-center"
          key="dob"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          <span>
            Date of birth
          </span>
          <span
            className="d-inline-flex flex-column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <ArrowUpIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 1184q0 13-10 23l-50 50q-10 10-23 10t-23-10l-393-393-393 393q-10 10-23 10t-23-10l-50-50q-10-10-10-23t10-23l466-466q10-10 23-10t23 10l466 466q10 10 10 23z"
                />
              </svg>
            </ArrowUpIcon>
            <ArrowDownIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 736q0 13-10 23l-466 466q-10 10-23 10t-23-10l-466-466q-10-10-10-23t10-23l50-50q10-10 23-10t23 10l393 393 393-393q10-10 23-10t23 10l50 50q10 10 10 23z"
                />
              </svg>
            </ArrowDownIcon>
          </span>
        </span>
      </div>
    </section>
  </DataTableHeader>
  <DataTableBody
    columns={
      Array [
        Object {
          "heading": "ID",
          "keyName": "id",
          "sortable": false,
          "width": "25%",
        },
        Object {
          "heading": "Name",
          "keyName": "name",
          "type": "string",
          "width": "25%",
        },
        Object {
          "heading": "Score",
          "keyName": "score",
          "type": "number",
          "width": "25%",
        },
        Object {
          "heading": "Date of birth",
          "keyName": "dob",
          "type": "date",
          "width": "25%",
        },
      ]
    }
    error={false}
    loading={false}
    source={
      Array [
        Object {
          "dob": "1990-02-02",
          "id": 1,
          "name": "Jack",
          "score": 200,
        },
        Object {
          "dob": "1980-12-02",
          "id": 2,
          "name": "Jack",
          "score": 200,
        },
        Object {
          "dob": "1996-09-22",
          "id": 3,
          "name": "Joe",
          "score": 100,
        },
        Object {
          "dob": "1990-02-02",
          "id": 4,
          "name": "Simon",
          "score": 400,
        },
        Object {
          "dob": "1988-05-11",
          "id": 5,
          "name": "Abraham",
          "score": 400,
        },
      ]
    }
  >
    <section>
      <div
        className="d-flex"
        key="0"
        style={
          Object {
            "padding": "1rem 0",
          }
        }
      >
        <span
          key="id"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          1
        </span>
        <span
          key="name"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          Jack
        </span>
        <span
          key="score"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          200
        </span>
        <span
          key="dob"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          1990-02-02
        </span>
      </div>
      <div
        className="d-flex"
        key="1"
        style={
          Object {
            "padding": "1rem 0",
          }
        }
      >
        <span
          key="id"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          2
        </span>
        <span
          key="name"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          Jack
        </span>
        <span
          key="score"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          200
        </span>
        <span
          key="dob"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          1980-12-02
        </span>
      </div>
      <div
        className="d-flex"
        key="2"
        style={
          Object {
            "padding": "1rem 0",
          }
        }
      >
        <span
          key="id"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          3
        </span>
        <span
          key="name"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          Joe
        </span>
        <span
          key="score"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          100
        </span>
        <span
          key="dob"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          1996-09-22
        </span>
      </div>
      <div
        className="d-flex"
        key="3"
        style={
          Object {
            "padding": "1rem 0",
          }
        }
      >
        <span
          key="id"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          4
        </span>
        <span
          key="name"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          Simon
        </span>
        <span
          key="score"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          400
        </span>
        <span
          key="dob"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          1990-02-02
        </span>
      </div>
      <div
        className="d-flex"
        key="4"
        style={
          Object {
            "padding": "1rem 0",
          }
        }
      >
        <span
          key="id"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          5
        </span>
        <span
          key="name"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          Abraham
        </span>
        <span
          key="score"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          400
        </span>
        <span
          key="dob"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          1988-05-11
        </span>
      </div>
    </section>
  </DataTableBody>
</DataTable>
`;

exports[`Test DataTable component snapshots should match snapshot of error 1`] = `
<DataTable
  columns={
    Array [
      Object {
        "heading": "ID",
        "keyName": "id",
        "sortable": false,
        "width": "25%",
      },
      Object {
        "heading": "Name",
        "keyName": "name",
        "type": "string",
        "width": "25%",
      },
      Object {
        "heading": "Score",
        "keyName": "score",
        "type": "number",
        "width": "25%",
      },
      Object {
        "heading": "Date of birth",
        "keyName": "dob",
        "type": "date",
        "width": "25%",
      },
    ]
  }
  error={true}
  loading={false}
  source={
    Array [
      Object {
        "dob": "1990-02-02",
        "id": 1,
        "name": "Jack",
        "score": 200,
      },
      Object {
        "dob": "1980-12-02",
        "id": 2,
        "name": "Jack",
        "score": 200,
      },
      Object {
        "dob": "1996-09-22",
        "id": 3,
        "name": "Joe",
        "score": 100,
      },
      Object {
        "dob": "1990-02-02",
        "id": 4,
        "name": "Simon",
        "score": 400,
      },
      Object {
        "dob": "1988-05-11",
        "id": 5,
        "name": "Abraham",
        "score": 400,
      },
    ]
  }
>
  <DataTableHeader
    columns={
      Array [
        Object {
          "heading": "ID",
          "keyName": "id",
          "sortable": false,
          "width": "25%",
        },
        Object {
          "heading": "Name",
          "keyName": "name",
          "type": "string",
          "width": "25%",
        },
        Object {
          "heading": "Score",
          "keyName": "score",
          "type": "number",
          "width": "25%",
        },
        Object {
          "heading": "Date of birth",
          "keyName": "dob",
          "type": "date",
          "width": "25%",
        },
      ]
    }
    onChangeSorting={[Function]}
    sortCriterias={Array []}
  >
    <section
      style={
        Object {
          "fontWeight": 500,
          "padding": "1rem 0",
        }
      }
    >
      <div
        className="d-flex"
      >
        <span
          className="d-flex justify-content-between align-items-center"
          key="id"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          <span>
            ID
          </span>
        </span>
        <span
          className="d-flex justify-content-between align-items-center"
          key="name"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          <span>
            Name
          </span>
          <span
            className="d-inline-flex flex-column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <ArrowUpIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 1184q0 13-10 23l-50 50q-10 10-23 10t-23-10l-393-393-393 393q-10 10-23 10t-23-10l-50-50q-10-10-10-23t10-23l466-466q10-10 23-10t23 10l466 466q10 10 10 23z"
                />
              </svg>
            </ArrowUpIcon>
            <ArrowDownIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 736q0 13-10 23l-466 466q-10 10-23 10t-23-10l-466-466q-10-10-10-23t10-23l50-50q10-10 23-10t23 10l393 393 393-393q10-10 23-10t23 10l50 50q10 10 10 23z"
                />
              </svg>
            </ArrowDownIcon>
          </span>
        </span>
        <span
          className="d-flex justify-content-between align-items-center"
          key="score"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          <span>
            Score
          </span>
          <span
            className="d-inline-flex flex-column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <ArrowUpIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 1184q0 13-10 23l-50 50q-10 10-23 10t-23-10l-393-393-393 393q-10 10-23 10t-23-10l-50-50q-10-10-10-23t10-23l466-466q10-10 23-10t23 10l466 466q10 10 10 23z"
                />
              </svg>
            </ArrowUpIcon>
            <ArrowDownIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 736q0 13-10 23l-466 466q-10 10-23 10t-23-10l-466-466q-10-10-10-23t10-23l50-50q10-10 23-10t23 10l393 393 393-393q10-10 23-10t23 10l50 50q10 10 10 23z"
                />
              </svg>
            </ArrowDownIcon>
          </span>
        </span>
        <span
          className="d-flex justify-content-between align-items-center"
          key="dob"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          <span>
            Date of birth
          </span>
          <span
            className="d-inline-flex flex-column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <ArrowUpIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 1184q0 13-10 23l-50 50q-10 10-23 10t-23-10l-393-393-393 393q-10 10-23 10t-23-10l-50-50q-10-10-10-23t10-23l466-466q10-10 23-10t23 10l466 466q10 10 10 23z"
                />
              </svg>
            </ArrowUpIcon>
            <ArrowDownIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 736q0 13-10 23l-466 466q-10 10-23 10t-23-10l-466-466q-10-10-10-23t10-23l50-50q10-10 23-10t23 10l393 393 393-393q10-10 23-10t23 10l50 50q10 10 10 23z"
                />
              </svg>
            </ArrowDownIcon>
          </span>
        </span>
      </div>
    </section>
  </DataTableHeader>
  <DataTableBody
    columns={
      Array [
        Object {
          "heading": "ID",
          "keyName": "id",
          "sortable": false,
          "width": "25%",
        },
        Object {
          "heading": "Name",
          "keyName": "name",
          "type": "string",
          "width": "25%",
        },
        Object {
          "heading": "Score",
          "keyName": "score",
          "type": "number",
          "width": "25%",
        },
        Object {
          "heading": "Date of birth",
          "keyName": "dob",
          "type": "date",
          "width": "25%",
        },
      ]
    }
    error={true}
    loading={false}
    source={
      Array [
        Object {
          "dob": "1990-02-02",
          "id": 1,
          "name": "Jack",
          "score": 200,
        },
        Object {
          "dob": "1980-12-02",
          "id": 2,
          "name": "Jack",
          "score": 200,
        },
        Object {
          "dob": "1996-09-22",
          "id": 3,
          "name": "Joe",
          "score": 100,
        },
        Object {
          "dob": "1990-02-02",
          "id": 4,
          "name": "Simon",
          "score": 400,
        },
        Object {
          "dob": "1988-05-11",
          "id": 5,
          "name": "Abraham",
          "score": 400,
        },
      ]
    }
  >
    <h5>
      Oops, something went wrong...
    </h5>
  </DataTableBody>
</DataTable>
`;

exports[`Test DataTable component snapshots should match snapshot of loading 1`] = `
<DataTable
  columns={
    Array [
      Object {
        "heading": "ID",
        "keyName": "id",
        "sortable": false,
        "width": "25%",
      },
      Object {
        "heading": "Name",
        "keyName": "name",
        "type": "string",
        "width": "25%",
      },
      Object {
        "heading": "Score",
        "keyName": "score",
        "type": "number",
        "width": "25%",
      },
      Object {
        "heading": "Date of birth",
        "keyName": "dob",
        "type": "date",
        "width": "25%",
      },
    ]
  }
  error={false}
  loading={true}
  source={
    Array [
      Object {
        "dob": "1990-02-02",
        "id": 1,
        "name": "Jack",
        "score": 200,
      },
      Object {
        "dob": "1980-12-02",
        "id": 2,
        "name": "Jack",
        "score": 200,
      },
      Object {
        "dob": "1996-09-22",
        "id": 3,
        "name": "Joe",
        "score": 100,
      },
      Object {
        "dob": "1990-02-02",
        "id": 4,
        "name": "Simon",
        "score": 400,
      },
      Object {
        "dob": "1988-05-11",
        "id": 5,
        "name": "Abraham",
        "score": 400,
      },
    ]
  }
>
  <DataTableHeader
    columns={
      Array [
        Object {
          "heading": "ID",
          "keyName": "id",
          "sortable": false,
          "width": "25%",
        },
        Object {
          "heading": "Name",
          "keyName": "name",
          "type": "string",
          "width": "25%",
        },
        Object {
          "heading": "Score",
          "keyName": "score",
          "type": "number",
          "width": "25%",
        },
        Object {
          "heading": "Date of birth",
          "keyName": "dob",
          "type": "date",
          "width": "25%",
        },
      ]
    }
    onChangeSorting={[Function]}
    sortCriterias={Array []}
  >
    <section
      style={
        Object {
          "fontWeight": 500,
          "padding": "1rem 0",
        }
      }
    >
      <div
        className="d-flex"
      >
        <span
          className="d-flex justify-content-between align-items-center"
          key="id"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          <span>
            ID
          </span>
        </span>
        <span
          className="d-flex justify-content-between align-items-center"
          key="name"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          <span>
            Name
          </span>
          <span
            className="d-inline-flex flex-column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <ArrowUpIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 1184q0 13-10 23l-50 50q-10 10-23 10t-23-10l-393-393-393 393q-10 10-23 10t-23-10l-50-50q-10-10-10-23t10-23l466-466q10-10 23-10t23 10l466 466q10 10 10 23z"
                />
              </svg>
            </ArrowUpIcon>
            <ArrowDownIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 736q0 13-10 23l-466 466q-10 10-23 10t-23-10l-466-466q-10-10-10-23t10-23l50-50q10-10 23-10t23 10l393 393 393-393q10-10 23-10t23 10l50 50q10 10 10 23z"
                />
              </svg>
            </ArrowDownIcon>
          </span>
        </span>
        <span
          className="d-flex justify-content-between align-items-center"
          key="score"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          <span>
            Score
          </span>
          <span
            className="d-inline-flex flex-column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <ArrowUpIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 1184q0 13-10 23l-50 50q-10 10-23 10t-23-10l-393-393-393 393q-10 10-23 10t-23-10l-50-50q-10-10-10-23t10-23l466-466q10-10 23-10t23 10l466 466q10 10 10 23z"
                />
              </svg>
            </ArrowUpIcon>
            <ArrowDownIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 736q0 13-10 23l-466 466q-10 10-23 10t-23-10l-466-466q-10-10-10-23t10-23l50-50q10-10 23-10t23 10l393 393 393-393q10-10 23-10t23 10l50 50q10 10 10 23z"
                />
              </svg>
            </ArrowDownIcon>
          </span>
        </span>
        <span
          className="d-flex justify-content-between align-items-center"
          key="dob"
          style={
            Object {
              "padding": "0 .5rem",
              "width": "25%",
            }
          }
        >
          <span>
            Date of birth
          </span>
          <span
            className="d-inline-flex flex-column"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <ArrowUpIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 1184q0 13-10 23l-50 50q-10 10-23 10t-23-10l-393-393-393 393q-10 10-23 10t-23-10l-50-50q-10-10-10-23t10-23l466-466q10-10 23-10t23 10l466 466q10 10 10 23z"
                />
              </svg>
            </ArrowUpIcon>
            <ArrowDownIcon
              opacity={0.5}
            >
              <svg
                height="16"
                style={
                  Object {
                    "opacity": 0.5,
                  }
                }
                viewBox="0 0 1792 1792"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1395 736q0 13-10 23l-466 466q-10 10-23 10t-23-10l-466-466q-10-10-10-23t10-23l50-50q10-10 23-10t23 10l393 393 393-393q10-10 23-10t23 10l50 50q10 10 10 23z"
                />
              </svg>
            </ArrowDownIcon>
          </span>
        </span>
      </div>
    </section>
  </DataTableHeader>
  <DataTableBody
    columns={
      Array [
        Object {
          "heading": "ID",
          "keyName": "id",
          "sortable": false,
          "width": "25%",
        },
        Object {
          "heading": "Name",
          "keyName": "name",
          "type": "string",
          "width": "25%",
        },
        Object {
          "heading": "Score",
          "keyName": "score",
          "type": "number",
          "width": "25%",
        },
        Object {
          "heading": "Date of birth",
          "keyName": "dob",
          "type": "date",
          "width": "25%",
        },
      ]
    }
    error={false}
    loading={true}
    source={
      Array [
        Object {
          "dob": "1990-02-02",
          "id": 1,
          "name": "Jack",
          "score": 200,
        },
        Object {
          "dob": "1980-12-02",
          "id": 2,
          "name": "Jack",
          "score": 200,
        },
        Object {
          "dob": "1996-09-22",
          "id": 3,
          "name": "Joe",
          "score": 100,
        },
        Object {
          "dob": "1990-02-02",
          "id": 4,
          "name": "Simon",
          "score": 400,
        },
        Object {
          "dob": "1988-05-11",
          "id": 5,
          "name": "Abraham",
          "score": 400,
        },
      ]
    }
  >
    <h5>
      Loading...
    </h5>
  </DataTableBody>
</DataTable>
`;
